llvm/lib/Target/X86/MCTargetDesc/X86BaseInfo.h

```diff
@@ -1182,11 +1182,27 @@ namespace X86II {
     }
   }
 
+  inline bool isXMMReg(unsigned RegNo) {
+    return (RegNo >= X86::XMM0 && RegNo <= X86::XMM15) ||
```

Comment:
Add comment to explain the reason, besides, add assert to make sure they are continuous, e.g.,

`assert(X86::XMM15 - X86::XMM0 == 15, "XMM registers are not continuous");`

Reply:
Besides, this will bring problems to code like this: https://github.com/llvm/llvm-project/blob/main/llvm/lib/Target/X86/X86RegisterInfo.cpp#L608

Reply:
We also updated the code there.